game/scripts/vscripts/components/duels/duels.lua

```diff
@@ -0,0 +1,309 @@
+-- Taken from bb template
+if Duels == nil then
+  DebugPrint ( 'Creating new Duels object.' )
+  Duels = class({})
+
+  Debug.EnabledModules['duels:*'] = true
+  Debug.EnabledModules['zonecontrol:*'] = true
+
+  ChatCommand:LinkCommand("-duel", "StartDuel", Duels)
+  ChatCommand:LinkCommand("-end_duel", "EndDuel", Duels)
+end
+
+function Duels:Init ()
+  DebugPrint('Init duels')
+
+  Duels.currentDuel = nil
+  Duels.zone1 = ZoneControl:CreateZone('duel_1', {
+    mode = ZONE_CONTROL_INCLUSIVE,
+    players = {
+    }
+  })
+
+  Duels.zone2 = ZoneControl:CreateZone('duel_2', {
+    mode = ZONE_CONTROL_INCLUSIVE,
+    players = {
+    }
+  })
+
+  GameEvents:OnHeroKilled(function (keys)
+    Duels:CheckDuelStatus(keys)
+  end)
+
+  Timers:CreateTimer(5, function ()
+    Duels:StartDuel()
+  end)
+end
+
+function Duels:CheckDuelStatus (keys)
+  if not Duels.currentDuel then
+    return
+  end
+
+  local playerId = keys.killed:GetPlayerOwnerID()
+  local foundIt = false
+
+  Duels:AllPlayers(Duels.currentDuel, function (player)
+    if foundIt or player.id ~= playerId then
+      return
+    end
+    foundIt = true
+    local scoreIndex = player.team .. 'Living' .. player.duelNumber
+    DebugPrint('Found dead player on ' .. player.team .. ' team with scoreindex ' .. scoreIndex)
+
+    Duels.currentDuel[scoreIndex] = Duels.currentDuel[scoreIndex] - 1
+
+    if Duels.currentDuel[scoreIndex] <= 0 then
+      Duels.currentDuel['duelEnd' .. player.duelNumber] = true
+      DebugPrint('Duel number ' .. scoreIndex .. ' is over and ' .. player.team .. ' lost')
+    end
+
+    if Duels.currentDuel.duelEnd1 and Duels.currentDuel.duelEnd2 then
+      DebugPrint('both duels are over, resuming normal play!')
+      Duels:EndDuel()
+    end
+  end)
+end
+
+function Duels:StartDuel ()
+  if Duels.currentDuel then
+    DebugPrint ('There is already a duel running')
+  end
```

Comment:
No, this is to make sure the duels don't overlap. they're run on timers and i want to avoid double timer bugs